src/components/atoms/Timeline/index.tsx

```diff
@@ -0,0 +1,245 @@
+      [epochDiff],
+    );
+
+    const startDate = useMemo(() => new Date(start), [start]);
+    const hoursCount = Math.trunc(HOURS_SECS / MEMORY_INTERVAL);
+    // Convert memory count to pixel.
+    const currentPosition = useMemo(() => {
+      const diff = Math.min((currentTime - start) / EPOCH_SEC / MEMORY_INTERVAL, memoryCount);
+      const strongMemoryCount = diff / (hoursCount * STRONG_MEMORY_HOURS);
+      return Math.max(
+        diff * GAP_HORIZONTAL +
+          (diff - strongMemoryCount) * NORMAL_MEMORY_WIDTH +
+          strongMemoryCount * STRONG_MEMORY_WIDTH +
+          STRONG_MEMORY_WIDTH / 2,
+        0,
+      );
+    }, [currentTime, start, memoryCount, hoursCount]);
+
+    return (
+      <Container>
+        <MemoryBox>
+          <MemoryList start={startDate} memoryCount={memoryCount} hoursCount={hoursCount} />
+          <Icon
+            style={{
+              left: currentPosition + PADDING_HORIZONTAL - knobSize / 2,
+            }}>
+            {renderKnob()}
+          </Icon>
+        </MemoryBox>
+      </Container>
+    );
+  },
+  (prev, next) => prev.currentTime === next.currentTime,
+);
+
+const MONTH_LABEL_LIST = [
+  "Jan",
+  "Feb",
+  "Mar",
+  "Apr",
+  "May",
+  "Jun",
+  "Jul",
+  "Aug",
+  "Sep",
+  "Oct",
+  "Nov",
+  "Dec",
+];
+
+type MemoryListProps = {
+  start: Date;
+  memoryCount: number;
+  hoursCount: number;
+};
+
+const MemoryList: React.FC<MemoryListProps> = memo(
+  function MemoryListPresenter({ start, memoryCount, hoursCount }) {
+    return (
+      <MemoryContainer>
+        {[...Array(memoryCount + 1)].map((_, idx) => {
+          const isHour = idx % hoursCount === 0;
+          const isStrongMemory = idx % (hoursCount * STRONG_MEMORY_HOURS) === 0;
+          if (isStrongMemory) {
+            const d = new Date(start.getTime() + idx * EPOCH_SEC * MEMORY_INTERVAL);
+            const year = d.getFullYear();
+            const month = MONTH_LABEL_LIST[d.getMonth()];
+            const date = `${d.getDate()}`.padStart(2, "0");
+            const hour = `${d.getHours()}`.padStart(2, "0");
+            // TODO: Fix date format
+            const label = `${month} ${date} ${year} ${hour}:00:00.0000`;
+
+            return (
+              <LabeledMemory>
+                <MemoryLabel>{label}</MemoryLabel>
+                <Memory key={`${idx}`} isHour={isHour} isStrongMemory={isStrongMemory} />
+              </LabeledMemory>
+            );
+          }
+          return <Memory key={`${idx}`} isHour={isHour} isStrongMemory={isStrongMemory} />;
+        })}
+      </MemoryContainer>
+    );
+  },
+  (prev, next) => prev.memoryCount === next.memoryCount && prev.start === next.start,
+);
+
+const Container = styled.div`
+  background: ${({ theme }) => theme.main.lighterBg};
+  width: 100%;
+  display: flex;
+  align-items: center;
+  padding: 5px 12px;
+  box-sizing: border-box;
+`;
+
+const MemoryBox = styled.div`
+  border: 1px solid #b5b5b5;
```

Comment:
Sorry, not sure I explained poorly above with Container's background color and using the theme. But, what I meant was all colors should use the theme. Make sure at some point you update all hardcoded colors!
Also, for most padding and margins you should make use of the theme's metrics.ts file (often for larger numbers a hardcode is fine (as there isn't a standard that @lavalse put in place above about 32px)

Reply:
I fixed in this comment https://github.com/reearth/reearth-web/pull/256#discussion_r912300818.